### PR TITLE
Standardize `FileAppend` fees to demonstrate fees refactor

### LINKED ISSUE #2072: Standardize `FileAppend` fee calculation
### Problem

`FileAppend` resource usage is estimated using a highly confusing functional style that creates unnecessary gRPC objects, and does not support generating an appropriate _feeSchedules.json_ entry for the base operation, given a price set by Product.

### Solution

- [x] Extracts the incremental resource usage in [`FileFeeBuilder.getFileAppendTxFeeMatrices()`](https://github.com/hashgraph/hedera-services/blob/master/hapi-utils/src/main/java/com/hederahashgraph/fee/FileFeeBuilder.java#L62) into a calculation supported by [`AccessorBasedUsages`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/fees/calculation/utils/AccessorBasedUsages.java#L43).
- [x] Updates `BaseOperationUsage` to support `FileAppend`, with the base resource usage matching that in _typedUsageFormulas.json_ [here](https://github.com/hashgraph/hedera-fee-tool-js/blob/master/src/resources/typedUsageFormulas.json#L1543).
- [x] Adds a `FileFeeSchedulesTest` method to validate the price for a `FileAppend` with this base usage is [exactly $0.05 as expected](https://docs.hedera.com/guides/mainnet/fees).
- [x] Updates [the `ScheduleGenerator.SUPPORTED_FUNCTIONS` list](https://github.com/hashgraph/hedera-services/blob/02072-M-StandardizeFileAppendFees/hapi-fees/src/main/java/com/hedera/services/pricing/ScheduleGenerator.java#L102) with `FileAppend`.
- [x] Replaces `FileAppend` prices in _feeSchedules.json_.
- [x] Updates `AccessorBasedUsages` to use `FileOpsUsage`.
- [x] Updates the EET client [`HapiFileAppend.feeFor()` implementation](https://github.com/hashgraph/hedera-services/blob/master/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/file/HapiFileAppend.java#L120) to use `FileOpsUsage.fileAppend()`.
- [x] Adds an EET spec to verify the price charged is as expected.
- [x] Removes legacy code in `FileFeeBuilder` and delete the `FileAppendResourceUsage` class.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #2072

**Notes for reviewer**:
ℹ️ &nbsp;This PR is a concrete example of the checklist for refactoring fees for a HAPI operation `X` (with `X=FileAppend`). Recall that the goals of this refactor are,

1. Support generating a _feeSchedules.json_ that includes `X`.
2. Eliminate unnecessary use of gRPC types in fee calculation for `X`.
3. Simplify the code for resource usage estimation for `X`.

The general checklist looks like:
- [x] Extracts the legacy resource estimate into a calculation in `AccessorBasedUsages`, typically based on one of the `*OpsUsage` .
- [x] Updates `BaseOperationUsage` to support `X`, with the base resource usage matching that in _typedUsageFormulas.json_ [here](https://github.com/hashgraph/hedera-fee-tool-js/blob/master/src/resources/typedUsageFormulas.json).
- [x] Adds a `*FeeSchedulesTest` method to validate the price for a `X` with this base usage is [as expected](https://docs.hedera.com/guides/mainnet/fees).
- [x] Updates the `ScheduleGenerator.SUPPORTED_FUNCTIONS` list with `X`.
- [x] Replaces `X` prices in _hedera-node/src/main/resources/feeSchedules.json_ and _test-clients/src/main/resource/FeeSchedule.json_.
- [x] Updates the EET client `HapiX.feeFor()` implementation to use the new .
- [x] Adds an EET spec to verify the base price charged is as expected.
- [x] Removes legacy code used for resource usage estimation.

### CODE REVIEW

File: hapi-fees/src/main/java/com/hedera/services/usage/file/FileAppendMeta.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+public class FileAppendMeta {
```
Comment (abhishek-hedera) [inline hapi-fees/src/main/java/com/hedera/services/usage/file/FileAppendMeta.java:23]: `FileAppendeMeta` could be made final.

**Review by abhishek-hedera — APPROVED** [review]
LGTM

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java [hunk 2]
```diff
@@ -169,17 +177,25 @@ UsageAccumulator baseUsageFor(HederaFunctionality function, SubType type) {
...
 			default:
 				break;
 		}
 
 		throw new IllegalArgumentException("Canonical usage unknown");
 	}
 
+	UsageAccumulator fileAppend() {
+		/* The canonical usage and context */
+		final var opMeta = new FileAppendMeta(1_000, THREE_MONTHS_IN_SECONDS);
```
Comment (ljianghedera) [inline hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:193]: The `FileAppendTransactionBody` doesn't have a field for (new) lifeTime. Does this imply that whenever a file has new appending contents, this file's expiry time will be automatically updated with the default value?

File: hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java [hunk 3]
```diff
@@ -65,6 +69,19 @@ static int bytesInBaseRepr() {
...
+			SigUsage sigUsage,
+			FileAppendMeta appendMeta,
+			BaseTransactionMeta baseMeta,
+			UsageAccumulator accumulator
+	) {
+		accumulator.resetForTransaction(baseMeta, sigUsage);
+
+		final var bytesAdded = appendMeta.getBytesAdded();
+		accumulator.addBpt(LONG_BASIC_ENTITY_ID_SIZE + bytesAdded);
+		accumulator.addSbs(bytesAdded * appendMeta.getLifetime());
```
Comment (ljianghedera) [inline hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java:82]: I am not sure how this works. This seems to me that different parts of this file may have different lifetime.  The newly appended part will have usage of "storage usage" * "this part's lifetime". What if the file's expiry time is earlier than the newly appended part's lifetime and the user doesn't renew or (auto-renew) when the file expires, does this mean the user is over-charged?

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java (hunk 2 above)
Comment (tinker-michaelj) [reply hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:193]: Right, the "lifetime" of the new contents is exactly the current lifetime of the file as it exists!

File: hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java (hunk 3 above)
Comment (tinker-michaelj) [reply hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java:82]: The lifetime of the newly appended part is exactly the remaining lifetime of the file as it currently exists.

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java (hunk 2 above)
Comment (ljianghedera) [reply hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:193]: I see. If the user appends new contents to an existing file, the services will charge the user of the added contents storage bytes size with the default "lifetime" value.

File: hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java (hunk 3 above)
Comment (ljianghedera) [reply hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java:82]: So does this mean the FileAppend transaction will change the file's remaining lifetime?

**Review by ljianghedera — APPROVED** [review]
Nice work!

**Pull request merged**